### PR TITLE
PSA: add restricted labels to openshift-gitops namespace

### PR DESCRIPTION
**Pull request by ibihim**
**What type of PR is this?**

/kind enhancement

**What does this PR do / why we need it**:

Add pod security policy restricted to the openshift-gitops namespace.

openshift-prefixed namespaces are not managed by the pod security policy sync controller, which sets pod security labels automatically for the users. It is expected that openshift namespaces consciously pick their security stance.

This means, that if pod security labels are not set, the defaults from the global config are being applied, which are set to restricted.

I am adding the SCC restricted-v2, such that the necessary SecurityContext should be set by SCC mutation.

It is related to the PR to upstream: https://github.com/argoproj-labs/argocd-operator/pull/1288.

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**Test acceptance criteria**:

* [ ] Unit Test
* [ ] E2E Test

**How to test changes / Special notes to the reviewer**:

- Workloads should work as before.
- `openshift-gitops` namespace should have the labels set in this PR.

### CODE REVIEW

**Review by iam-veeramalla — DISMISSED** [review]
/lgtm
/approve

File: controllers/gitopsservice_controller.go [hunk 1]
```diff
@@ -811,14 +815,26 @@ func newBackendService(ns types.NamespacedName) *corev1.Service {
...
 			// Enable full-fledged support for integration with cluster monitoring.
 			"openshift.io/cluster-monitoring": "true",
 		},
 	}
+
+	if strings.HasPrefix(ns, "openshift-") {
+		// Set pod security policy, which is required for namespaces pre-fixed with openshift
+		// as the pod security label syncer doesn't set them on OCP namespaces.
+		objectMeta.Labels["pod-security.kubernetes.io/enforce"] = "restricted"
+		objectMeta.Labels["pod-security.kubernetes.io/enforce-version"] = "v1.29"
```
Comment (svghadi) [inline controllers/gitopsservice_controller.go:831]: Hi @ibihim, just trying to understand this change. Was there any special reason to set it to v1.29 and not latest?

**Comment by ibihim** [comment]
Hi, yes. latest is better. at the moment in time, we used a fixed version in openshift, but we moved now ahead to latest.

**Comment by svghadi** [comment]
Thanks for the confirmation @ibihim . Will update it to latest.

**Pull request merged**